cpp/open3d/t/geometry/TriangleMeshFactory.cpp

```diff
@@ -0,0 +1,100 @@
+namespace geometry {
+
+TriangleMesh TriangleMesh::CreateBox(double width,
+                                     double height,
+                                     double depth,
+                                     core::Dtype float_dtype,
+                                     core::Dtype int_dtype,
+                                     const core::Device &device) {
+    // Check width, height, depth.
+    if (width <= 0) {
+        utility::LogError("width <= 0");
+    }
+    if (height <= 0) {
+        utility::LogError("height <= 0");
+    }
+    if (depth <= 0) {
+        utility::LogError("depth <= 0");
+    }
+
+    // Vertices.
+    core::Tensor vertex_positions =
+            core::Tensor::Init<double>({{0.0, 0.0, 0.0},
+                                        {width, 0.0, 0.0},
+                                        {0.0, 0.0, depth},
+                                        {width, 0.0, depth},
+                                        {0.0, height, 0.0},
+                                        {width, height, 0.0},
+                                        {0.0, height, depth},
+                                        {width, height, depth}},
+                                       device);
+
+    if (float_dtype == core::Float32) {
+        vertex_positions = vertex_positions.To(core::Float32);
+    } else if (float_dtype != core::Float64) {
+        utility::LogError("float_dtype must be Float32 or Float64, but got {}.",
+                          float_dtype.ToString());
+    }
+
+    // Triangles.
+    core::Tensor triangle_indices = core::Tensor::Init<int64_t>({{4, 7, 5},
+                                                                 {4, 6, 7},
+                                                                 {0, 2, 4},
+                                                                 {2, 6, 4},
+                                                                 {0, 1, 2},
+                                                                 {1, 3, 2},
+                                                                 {1, 5, 7},
+                                                                 {1, 7, 3},
+                                                                 {2, 3, 7},
+                                                                 {2, 7, 6},
+                                                                 {0, 4, 1},
+                                                                 {1, 4, 5}},
+                                                                device);
+
+    if (int_dtype == core::Int32) {
+        triangle_indices = triangle_indices.To(core::Int32);
+    } else if (int_dtype != core::Int64) {
+        utility::LogError("int_dtype must be Int32 or Int64, but got {}.",
+                          int_dtype.ToString());
+    }
+
+    // Mesh.
+    TriangleMesh mesh(vertex_positions, triangle_indices);
+
+    return mesh;
+}
+
+}  // namespace geometry
+}  // namespace t
+}  // namespace open3d
```

Comment:
always leave a newline at the end of the code. There is also some setting to do that in vs-code. https://stackoverflow.com/a/44704969

Reply:
ok, got it